### PR TITLE
@eessex => Infinite scroll + pull in latest reaction components

### PR DESCRIPTION
**Pull request by kanaabe**
Adds infinite scroll to the page using a new dependency, [`react-waypoint`](https://github.com/brigade/react-waypoint).

### CODE REVIEW

File: desktop/apps/article2/components/InfiniteScrollArticle.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      offset: 0,
+      error: false
+    }
+  }
+
+  fetchNextArticles = async () => {
+    this.setState({ isLoading: true })
+    try {
+      const data = await positronql({ query: articlesQuery(this.state.offset, 3, sd.ARTSY_EDITORIAL_CHANNEL) })
+      // Omit original article from infinite scroll
```
Comment (kanaabe) [inline desktop/apps/article2/components/InfiniteScrollArticle.js]: #TODO support it in the API

File: desktop/apps/article2/queries/article.js [hunk 2]
```diff
@@ -1,124 +1,18 @@
...
-        slug
+        ${articleBody}
       }
-      artists {
-        name
-        slug
-      }
-      artist {
-        name
+      relatedArticles: articles(published: true, limit: 3) {
```
Comment (kanaabe) [inline desktop/apps/article2/queries/article.js]: This query will be replaced with real API support.

File: desktop/apps/article2/components/InfiniteScrollArticle.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        offset: this.state.offset + 3
+      })
+    } catch (error) {
+      this.setState({ isLoading: false, error: true })
+    }
+  }
+
+  renderWaypoint = () => {
+    if (!this.state.isLoading) {
+      return <Waypoint onEnter={this.fetchNextArticles} threshold={2.0} />
```
Comment (eessex) [inline desktop/apps/article2/components/InfiniteScrollArticle.js]: Ooh react-waypoints, very exciting

File: desktop/apps/article2/components/InfiniteScrollArticle.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        return article.id !== this.props.article.id
+      })
+      this.setState({
+        articles: this.state.articles.concat(articles),
+        isLoading: false,
+        offset: this.state.offset + 3
+      })
+    } catch (error) {
+      this.setState({ isLoading: false, error: true })
+    }
```
Comment (eessex) [inline desktop/apps/article2/components/InfiniteScrollArticle.js]: This spot could definitely be a use case for Redux

**Comment by eessex** [comment]
Looks good from here! Merging.

**Pull request merged**